utils/docker-network-ipaddresspool.sh

```diff
@@ -13,16 +13,21 @@ YQ="${2:-yq}"
 
 ## Parse kind network subnet
 ## Take only IPv4 subnets, exclude IPv6
-SUBNET=$(docker network inspect $networkName --format '{{json .IPAM.Config }}' | \
-    ${YQ} '.[] | select( .Subnet | test("^((25[0-5]|(2[0-4]|1\d|[1-9]|)\d)\.?\b){4}/\d+$")) | .Subnet')
+SUBNET=""
+if command -v podman &> /dev/null; then
+  SUBNET=`podman network inspect -f '{{range .Subnets}}{{if eq (len .Subnet.IP) 4}}{{.Subnet}}{{end}}{{end}}' $networkName`
+elif command -v docker &> /dev/null; then
+  SUBNET=`docker network inspect $networkName -f '{{ (index .IPAM.Config 0).Subnet }}'`
+fi
```

Comment:
Is this not assuming if i have podman installed that i want to use it? I have both installed but use docker, this will try and inspect the podman network first and fail no?

Reply:
I think that's why the CI job is failing actually.

If you have both installed, and don't have the 'kind' network setup in podman, it would give that error.
If you have just podman installed and a docker alias, this would work fine as all other commands use 'podman'.
If you have just podman installed and no alias for docker, the setup would fail a lot earlier.

If we switched to the docker way of doing things first, that would fail for anyone with just podman & the docker alias.
Maybe that's the right ordering though, and have it fallback to the podman version of the command if there's an error?

Reply:
I've made another change to try podman, then docker, then fail

Reply:
I think this might work as long as you have never created the kind network using both podman and docker. On a Mac is it expected that you do not have docker installed? When you create a kind cluster are you telling it what container runtime to use or relying on it to auto detect?

This change is probably fine, but people should probably be aware that it's being made and that if you happen to have used both podman and docker at any time, especially with kind, that you could get odd results.